frontend/src/toolbar/shared/heatmapLogic.js

```diff
@@ -0,0 +1,277 @@
+            false,
+            {
+                setHeatmapEnabled: (_, { heatmapEnabled }) => heatmapEnabled,
+            },
+        ],
+        clicks: [
+            0,
+            {
+                addClick: state => state + 1,
+            },
+        ],
+        highlightedElement: [
+            null,
+            {
+                highlightElement: (_, { element }) => element,
+                setHeatmapEnabled: (state, { heatmapEnabled }) => (heatmapEnabled ? state : null),
+                [inspectElementLogic.actions.start]: () => null,
+            },
+        ],
+        selectedElement: [
+            null,
+            {
+                selectElement: (state, { element }) => (state === element ? null : element),
+                setHeatmapEnabled: (state, { heatmapEnabled }) => (heatmapEnabled ? state : null),
+                [inspectElementLogic.actions.start]: () => null,
+            },
+        ],
+        showElementFinder: [
+            false,
+            {
+                highlightElement: (_, { withElementFinder }) => withElementFinder,
+                setHeatmapEnabled: (state, { heatmapEnabled }) => (heatmapEnabled ? state : false),
+                [inspectElementLogic.actions.start]: () => false,
+            },
+        ],
+        heatmapLoading: [
+            false,
+            {
+                getEvents: () => true,
+                getEventsSuccess: () => false,
+                getEventsFailure: () => false,
+                resetEvents: () => false,
+            },
+        ],
+    }),
+
+    loaders: () => ({
+        events: [
+            [],
+            {
+                resetEvents: () => [],
+                getEvents: async ({ $current_url }, breakpoint) => {
+                    const params = {
+                        properties: [{ key: '$current_url', value: $current_url }],
+                        temporary_token: toolbarLogic.values.temporaryToken,
+                    }
+                    const url = `${toolbarLogic.values.apiURL}api/element/stats/${encodeParams(params, '?')}`
+                    const response = await fetch(url)
+                    const results = await response.json()
+
+                    if (response.status === 403) {
+                        toolbarLogic.actions.authenticate()
+                        return []
+                    }
+
+                    breakpoint()
+
+                    if (!Array.isArray(results)) {
+                        throw new Error('Error loading HeatMap data!')
+                    }
+
+                    return results
+                },
+            },
+        ],
+    }),
+
+    selectors: ({ selectors }) => ({
+        elements: [
+            () => [selectors.events],
+            events => {
+                const elements = events
+                    .map(event => {
+                        let combinedSelector
+                        for (let i = event.elements.length - 1; i >= 0; i--) {
```

Comment:
I think something might be going wrong with the ordering here. In my case, it thinks the clicked element is the body, despite that being the last element in the list:
![image](https://user-images.githubusercontent.com/1727427/84646372-4a537400-af02-11ea-8931-2f6d67f24a18.png)

Reply:
That's really weird. For all the elements I get back from my local API, `body` is the first tag. 
![image](https://user-images.githubusercontent.com/53387/84753267-0ffbdc80-afbf-11ea-9801-63997642f185.png)

I added a patch that reverses the elements if `body` is in the wrong end of the list... though I feel uncomfortable not knowing why this discrepancy exists.

Reply:
We've had this before, I just had to add a specific `.order_by('order')` to the queryset. I've done that now and pushed that up to this branch so you can remove that hack

Reply:
Thanks, that fixed this. However after the change the number of elements I'm getting from the API dropped dramatically. Now it returns just 1 element vs 3 before... plus 10 that were not found on the page (different app on localhost:3000).

I guess this `order` field is not filled in for old data. I guess this is fine and this has been enabled already for a while in production?